presto-main/src/main/java/com/facebook/presto/sql/rewrite/DescribeInputRewrite.java

```diff
@@ -0,0 +1,145 @@
+import static com.facebook.presto.sql.QueryUtil.ordering;
+import static com.facebook.presto.sql.QueryUtil.row;
+import static com.facebook.presto.sql.QueryUtil.selectList;
+import static com.facebook.presto.sql.QueryUtil.simpleQuery;
+import static com.facebook.presto.sql.QueryUtil.values;
+import static com.facebook.presto.type.UnknownType.UNKNOWN;
+import static java.util.Objects.requireNonNull;
+
+final class DescribeInputRewrite
+        implements StatementRewrite.Rewrite
+{
+    @Override
+    public Statement rewrite(
+            Session session,
+            Metadata metadata,
+            SqlParser parser,
+            Optional<QueryExplainer> queryExplainer,
+            Statement node,
+            List<Expression> parameters,
+            AccessControl accessControl,
+            boolean experimentalSyntaxEnabled)
+    {
+        return (Statement) new Visitor(session, parser, metadata, queryExplainer, parameters, accessControl, experimentalSyntaxEnabled).process(node, null);
+    }
+
+    private static final class Visitor
+            extends AstVisitor<Node, Void>
+    {
+        private final Session session;
+        private final SqlParser parser;
+        private final Metadata metadata;
+        private final Optional<QueryExplainer> queryExplainer;
+        private final List<Expression> parameters;
+        private final AccessControl accessControl;
+        private final boolean experimentalSyntaxEnabled;
+
+        public Visitor(
+                Session session,
+                SqlParser parser,
+                Metadata metadata,
+                Optional<QueryExplainer> queryExplainer,
+                List<Expression> parameters,
+                AccessControl accessControl,
+                boolean experimentalSyntaxEnabled)
+        {
+            this.session = requireNonNull(session, "session is null");
+            this.parser = parser;
+            this.metadata = metadata;
+            this.queryExplainer = queryExplainer;
+            this.accessControl = accessControl;
+            this.parameters = parameters;
+            this.experimentalSyntaxEnabled = experimentalSyntaxEnabled;
+        }
+
+        @Override
+        protected Node visitDescribeInput(DescribeInput node, Void context)
+                throws SemanticException
+        {
+            String sqlString = session.getPreparedStatement(node.getName());
+            Statement statement = parser.createStatement(sqlString);
+
+            // create  analysis for the query we are describing.
+            Analyzer analyzer = new Analyzer(session, metadata, parser, accessControl, queryExplainer, experimentalSyntaxEnabled, parameters);
+            Analysis analysis = analyzer.analyze(statement, true);
+
+            // get all parameters in query
+            List<Parameter> parameters = getParameters(statement);
+
+            // return the positions and types of all parameters.  If there are no parameters, return a single null row.
```

Comment:
No. I initially did it that way because there was no way to put 0 rows into a values statement because then you don't know what your fields are, and a null row seemed to sufficiently convey the same information.  I'm realizing now that I can just add a limit 0 for a case with no parameters.